### ISSUE TITLE
About Hook SAPHanaSR

### ISSUE THREAD

**Issue by ScaMar** [SUSE/SAPHanaSR#35]
Hello, i am configuring the Hook SAPHanaSR on SLES12 SP3, with a SAP HANA `1.00.122.20.1536720790.`
Such sections have been added to global.ini
```
[ha_dr_provider_SAPHanaSR]
provider = SAPHanaSR
path = /usr/share/SAPHanaSR
execution_order = 1

[trace]
ha_dr_saphanasr = info
```
Such line in the sudoers file:
`h1padm ALL=(ALL) NOPASSWD: /usr/sbin/crm_attribute -n hana_h1p_site_srHook_*`

When i try to start the HANA instance, it ends in a loop with these messages:
```
[75888]{-1}[-1/-1] 2020-01-31 15:03:43.061605 i ha_dr_provider   PythonProxyImpl.cpp(01052) : calling HA/DR provider SAPHanaSR.hookDRConnectionChanged(hostname=lnxhdb120pri, port=30001, database=SYSTEMDB, status=12, database_status=12, system_status=12, timestamp=2020-01-31T15:03:43.061563+01:00, is_in_sync=0, reason=Starting)
[75888]{-1}[-1/-1] 2020-01-31 15:03:43.061744 i ha_dr_SAPHanaSR  SAPHanaSR.py(00086) : SAPHanaSR (0.161.0) SAPHanaSR.srConnectionChanged method called with Dict={'status': 12, 'is_in_sync': False, 'database': 'SYSTEMDB', 'timestamp': '2020-01-31T15:03:43.061563+01:00', 'hostname': 'lnxhdb120pri', 'system_status': 12, 'reason': 'Starting', 'database_status': 12, 'port': '30001'}
[75888]{-1}[-1/-1] 2020-01-31 15:03:43.061787 e ha_dr_provider   PythonProxyImpl.cpp(01099) : SAPHanaSR/SAPHanaSR:srConnectionChanged() failed with python error: 'siteName'
  SAPHanaSR.py(94): mySite = ParamDict["siteName"]
[75443]{-1}[-1/-1] 2020-01-31 15:03:43.061827 e sr_nameserver    DisasterRecoveryPrimaryCallbackImpl.cpp(00181) : Could not send connectionEstablished event to HA/DR Provider. Retry in 5 seconds.
```
Is this a known issue? Am i doing something wrong?

Regards,
Marco

**Comment by fmherschel** [comment]
For a full support please open a ticket either a SUSE SR or an SAP support ticket.
Which SAPHanaSR python hook version did you have installed - the one from the package?
And which SAPHanaSR* package version do you have installed? Are you using Scale-Out or Scale-Up?

**Comment by fmherschel** [comment]
I also see that you have still a SAP HANA 1.0 (rev 122) installed. The hook was never planned for
that old SAP HANA it was integrated into SAP HANA 2.0. Maybe SAP did some changes on the
level of the HA/DR/SR provider API I would need to reproduce that.
If I can reproduce the error I will report that back here and will try to get a solution for this.

**Comment by ScaMar** [comment]
The scenario is a Scale-Up one. The hook is the one provided with SAPHanaSR-0.153.2-3.8.2.noarch rpm package. Unluckly HANA can't be upgraded because on top of it there is an application server (non SAP) that do not (still) support HANA 2. Thank you for your feedback and your effort, the hook is very nice to have technology.

**Comment by fmherschel** [comment]
Yes - give me some time till I could start to reproduce that. If you need to speed-up that I would like to open a SUSE or SAP ticket. Thanks for reporting that issue.

**Comment by fmherschel** [comment]
The main difference between HANA 1.0 and 2.0 is the multi tenancy (MDC). Maybe this is the tricky part for your software, because now you need to specify which database inside HANA you need to contact.

**Comment by ScaMar** [comment]
I have no hurry, if you like/prefer i can open a SUSE or SAP ticket. According the topology, the HANA instance has been set up as multi container. Actually there are two tenants, id1 ->  SYSTEMDB and id3 -> PFN.

**Comment by fmherschel** [comment]
Ok sorry for the delay. 

I think I got what went wrong here. It seams that SAP HANA 1.0 rev 122 had a different provider
interface for system replication errors and the site-name is missing in the python dictionary.
In this case the hook included in the package is only compatible with SAP HANA 2.0.

If you need to use the hook, please open a SAP ticket on the SUSE component BC-OP-LNX-SUSE and also refer to this issue (URL). I hope my support colleagues will then creating a formal SUSE
bugzilla report. Sorry I would like to use the official processes for such a (bigger) change as it
could be critical to adapt the hook to be now backward compatible with SAP HANA 1.0.

Maybe we need to do this with an other hook script - not to break the already implemented clusters with SAP HANA 2.0. If the interfaces are differ on that level I would not like to hide this in one single
hook script at this could drastic increase the complexity of the scripts.

**Comment by ScaMar** [comment]
Hello,
thank you for your support and your dedication. The hook isn't needed anymore because the customer moved the istances into virtual environments, with dedicated virtual machines. According to your proposal, looking at end of maintenance for HANA 1 (May 2021, extended from 2019 already), i think the result do not fit the effort needed to implement such feature on HANA 1. I will not ask for it. Thank you again.
